In [9]:
# import the packages that are required for Data Wrangling.

import pandas as pd
import numpy as np

In [18]:
# Read the creditcard.csv file.

df = pd.read_csv('creditcard.csv')

In [16]:
print ('Total Rows :', df.shape[0])
print ('Total Columns :', df.shape[1])

Total Rows : 284807
Total Columns : 31


In [22]:
print ('Column Names : ', df.columns.tolist())

Column Names :  ['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10', 'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20', 'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount', 'Class']


In [ ]:
print ('Data types of each column', df.info())

In [ ]:
print ('Data Description of each column : ', df.describe())

In [35]:
print ('Null Values in the Dataset :', df.isnull().values.any())

Null Values in the Dataset : False


1) What kind of cleaning steps did you perform?
This Dataset is already cleaned. And there is no null values in any of the columns. Hence No data cleaning
is required.

2) How did you deal with missing values, if any?

Case 1 : Out of 284807 rows, if the Null values are below 1000 rows. We can drop those rows.

Case 2 : Suppose, If in case this data has got null or NaN values, which more than 1000 rows,
          Column : Time - We can forward fill the data here, Because this columns contain the time of transactions
             starting from 0. And the values are in secs between first and consecutive transaction. And there is
             a possibility of having two or more transactions happening at same time, different places.
         
   V1, V2...V28 - This is a masked data of transactions, PCA transformed. If any values are missing here. We can 
                  take the mean value of missing column and fill in that particular cell. But we have make sure 
                  that there are no Outliers.
   
   Amount - This column contains the amount in euros. We can fill this values by taking mean of Fraud or Non-fraud
            based on the class type.
            
   Class - Ask mentor. And what if any Categorical values are there.
                
3) Were there outliers, and how did you handle them?
    
Ask mentor.